### ISSUE TITLE
poll(socket, "recv") returns true but recv(socket) returns None

### ISSUE THREAD

**Issue by nehpetsde** [nfcpy/nfcpy#5]
I ran into a bug while using the NPP server. It uses this code to read data:

            while nfc.llcp.poll(socket, "recv"):
                data += nfc.llcp.recv(socket)

But I got this exception:

Exception in thread Thread-3:
Traceback (most recent call last):
  File "/usr/local/Python/lib/python2.6/threading.py", line 525, in __bootstrap_inner
    self.run()
  File "/usr/local/Python/lib/python2.6/threading.py", line 477, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/Users/andre/Dropbox/cmu/thesis/code/vmachine/nfcpy/nfc/npp/server.py", line 47, in serve
    data += nfc.llcp.recv(socket)
TypeError: cannot concatenate 'str' and 'NoneType' objects

So it seems that poll() returned true, so there was something to read, but recv() returned None.

------------------------------------
Imported from Launchpad using lp2gh.

 * date created: 2011-06-11T16:51:37Z
 * owner: andrefcruz
 * assignee: stephen-tiedemann
 * the launchpad url was https://bugs.launchpad.net/bugs/795975

**Comment by nehpetsde** [comment]
(by stephen-tiedemann)
I could reproduce the problem and shall have a bugfix soon.

**Comment by nehpetsde** [comment]
(by stephen-tiedemann)
Fixed with trunk revisions r68 and r69

**Comment by nehpetsde** [comment]
(by stephen-tiedemann)
Andre, I've also made some smaller changes to the NPP server to reflect the fix to poll("recv"). But more important to note might be that I've changed the signature of process(), which now provides the ndef message as binary string (i.e. the custom code derived from npp server must run the data through the ndef parser. Reason being that there might be situations where the data shall go unmodified (or in any other forma than a parsed ndef message) to an external process.
